### ISSUE TITLE
Support jupyterlab3.0+ (also the release candidate?)

### ISSUE THREAD

**Issue by cnydw** [jupyterlab/jupyterlab-hdf5#42]
JupyterLab is going to release 3.0 version soon, I tried to use the hdf5 extension with jupyterlab==3.0.0rc6, and was able to make it working locally by updating the dependencies in `package.json`.

```
  "dependencies": {
    "@jupyterlab/application": "^3.0.0-rc.6",
    "@jupyterlab/apputils": "^3.0.0-rc.6",
    "@jupyterlab/coreutils": "^5.0.0-rc.6",
    "@jupyterlab/docmanager": "^3.0.0-rc.6",
    "@jupyterlab/docregistry": "^3.0.0-rc.6",
    "@jupyterlab/filebrowser": "^3.0.0-rc.6",
    "@jupyterlab/notebook": "^3.0.0-rc.6",
    "@jupyterlab/services": "^6.0.0-rc.6",
    "@lumino/algorithm": "^1.3.0",
    "@lumino/coreutils": "^1.5.0",
    "@lumino/datagrid": "^0.14.0",
    "@lumino/messaging": "^1.4.0",
    "@lumino/signaling": "^1.4.0",
    "@lumino/widgets": "^1.14.0"
  },
  "devDependencies": {
    "husky": "^3.0.1",
    "lint-staged": "^9.2.0",
    "prettier": "^1.13.7",
    "rimraf": "~2.6.2",
    "tslint": "^5.10.0",
    "tslint-config-prettier": "^1.13.0",
    "tslint-plugin-prettier": "^2.0.1",
    "typescript": "~4.0.2",
    "yarn-deduplicate": "^1.1.1"
  },
```

Do we wait until the official release of jupyterlab3.0 to publish the corresponding npm package? Or maybe we can already publish a corresponding release candidate version?

**Comment by mtracy-core** [comment]
Wondering if this is still in progress? or if upgrading to 3 has been completed?

**Comment by loichuder** [comment]
Thanks @cnydw, I was able to make it work as well on jupyterlab 3.0.5 with the updated deps: https://github.com/jupyterlab/jupyterlab-hdf5/pull/73

Waiting on @telamonian input for the release plan.

**Comment by telamonian** [comment]
@loichuder Let's wrap up the current batch of PRs, cut a "final" (critical fixes can always be backported) jlab 2.x compatible release, then do the 3.x upgrade

**Comment by wbolton** [comment]
Hey, was wondering if this is still in progress?

**Comment by loichuder** [comment]
It is !

We have still a few ongoing issues to solve before the next JupyterLab 2 release. That should be shortly sorted out.

Then, we will implement the JupyterLab 3 support that should hopefully be as simple as merging #73.

**Comment by meeseeksmachine** [comment]
This issue has been mentioned on **Jupyter Community Forum**. There might be relevant details there:

https://discourse.jupyter.org/t/jupyterlab-hdf5-update-for-jupyterlab-3/7598/2

**Comment by jabbera** [comment]
Hi!

Any chance we can get this merged soon?

Cheers,
Mike

**Comment by apd47** [comment]
Hi - wanted to check if there were any plans for a release or roadblocks I can help with. We're about to bump a few Jlab Dockerfiles over to 3.x and this extension will very soon be blocking the upgrade since our organization requires use of a formal release candidate. @loichuder, I noticed there hasnt been another 2.x release either - how can we help push things through?

**Comment by blink1073** [comment]
@loichuder what is your npm username?

**Comment by loichuder** [comment]
[loichuder](https://www.npmjs.com/~loichuder) on npm. 

But I guess PyPI would also be needed for `jupyterlab_hdf`: [huderl](https://pypi.org/user/huderl/) on PyPI

**Comment by blink1073** [comment]
Invites sent!

**Comment by aufdenkampe** [comment]
What is the timeline for merging PR #106 and supporting JupyterLab v3?

**Comment by loichuder** [comment]
I will do a last 2.x release this month to add the support hard and soft links.

I hope to have a 3.x release landing at the end of the month :crossed_fingers:

**Comment by loichuder** [comment]
Support of JupyterLab3 landed with v1.0.0
